frame/staking-rewards/src/lib.rs

```diff
@@ -180,6 +191,17 @@ pub mod pallet {
 		>,
 	>;
 
+    /// Abstraction over Stake type
+    type StakeOf<T> = Stake<
+        <T as Config>::RewardPoolId,
+        <T as Config>::Balance,
+        Rewards<
```

Comment:
Let's introduce a new type for this called `Reductions` for this with a basic map with the same generic params